src/event_pusher/mod_event_pusher_rabbit.erl

```diff
@@ -85,6 +96,53 @@ publish_user_presence_change(JID, Status) ->
                                    exchange => PresenceExchange}},
                available_worker).
 
+-spec publish_user_chat_message_sent_event(From :: jid:jid(),
+                                           To :: jid:jid(),
+                                           Packet :: exml:element()) -> ok.
+publish_user_chat_message_sent_event(From = #jid{lserver = Host}, To, Packet) ->
+    Topic = opt(Host, chat_msg_sent_topic, ?DEFAULT_CHAT_MSG_SENT_TOPIC),
+    publish_user_chat_message_event(user_chat_msg_sent, #{from => From,
+                                                          to => To,
+                                                          packet => Packet,
+                                                          topic => Topic}).
+
+-spec publish_user_chat_message_received_event(From :: jid:jid(),
+                                               To :: jid:jid(),
+                                               Packet :: exml:element()) -> ok.
+publish_user_chat_message_received_event(From = #jid{lserver = Host}, To,
+                                         Packet) ->
+    Topic = opt(Host, chat_msg_recv_topic, ?DEFAULT_CHAT_MSG_RECV_TOPIC),
+    publish_user_chat_message_event(user_chat_msg_recv, #{from => From,
+                                                          to => To,
+                                                          packet => Packet,
+                                                          topic => Topic}).
+-spec publish_user_chat_message_event(Event :: user_chat_msg_sent
+                                             | user_chat_msg_recv, map()) -> ok.
+publish_user_chat_message_event(Event, #{from := From = #jid{lserver = Host},
+                                         to := To,
+                                         packet := Packet,
+                                         topic := Topic}) ->
+    FromJID = jid_from_chat_event(From),
+    ToJID = jid_from_chat_event(To),
+    Message = extract_message(Packet),
+    ChatMsgExchange = opt(Host, chat_msg_exchange, ?DEFAULT_CHAT_MSG_EXCHANGE),
+    wpool:cast(pool_name(Host), {Event, #{from_jid => FromJID,
+                                          to_jid => ToJID,
+                                          msg => Message,
+                                          exchange => ChatMsgExchange,
+                                          topic => Topic}},
+               available_worker).
+
+-spec jid_from_chat_event(JID :: jid:jid()) -> binary().
+jid_from_chat_event(JID) ->
```

Comment:
Function name is misleading. Actually it is `jid_to_lower_binary`. Also it may be reused in `publish_user_presence_change`.

Reply:
👍